woocommerce/woocommerce
# Replace flaky Local Pickup e2e test with PHP and JS unit test

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Removes the e2e test and replaces it with a JS unit test and a PHP unit test.

The JS unit test is to ensure we still _get_ the local pickup title from the asset data registry, this will catch changes when the JS block is updated, but won't catch the PHP side, that is covered in the PHP unit test.

In the PHP unit test, I instantiate a new `CheckoutBlock` this is to re-add data to the `AssetDataRegistry` and to allow the contents to be observed (by using the mock).

In the PHP unit test, the Checkout block being saved with a changed `localPickupText` is tested and the value saved to options is verified.



Closes #48667 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure CI is passing.
2. Read through the test changes and ensure they're testing/checking the right things compared to the old E2E test.(updating the title in WC Settings updates the local pickup text in the block and vice/versa).



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
